server/aesdsocket.c

```diff
@@ -358,284 +357,280 @@ static void handle_socket()
+        { 
+          if(buffer[i] == '\n')  // check at end of packet
+            {
+              is_packet_received = true;    //end of packet detected
+              i++;          
+              break;
+            }
         }
-      else if(file_write != strlen(output_buffer))
+      received_packet_size += i; //increment the size of packet
+      output_buffer = (char *) realloc(output_buffer,(received_packet_size+1)); //reloaction to a buffer big in size
+      if(output_buffer == NULL)
         {
-          printf("Error! File is partially written\n");
-          syslog(LOG_ERR,"File is partially written");
+          printf("Realloc() has failed\n");
+          syslog(LOG_ERR, "Realloc has failed!\n");
           exit(1);
         }
       else
         {
-          printf("File is written successfully\n");
-          syslog(LOG_INFO,"File is written successfully");
+          printf("Realloc() succeed\n");
+          syslog(LOG_INFO, "Realloc succeed\n");
         }
-        
-        printf("Writing data to file %s \n",output_buffer);
+      //printf("buffer = %s \n",buffer);
+      strncat(output_buffer,buffer,i);
+      memset(buffer,0,BUFFER_SIZE); 
+    }
 
-      close(file_fd);
+  mutex_return_val = pthread_mutex_lock(params->mutex);
+  if (mutex_return_val != 0)
+    {
+      printf("pthread_mutex_lock() has failed\n");
+      syslog(LOG_ERR, "pthread_mutex_lock() has failed\n");
+      exit(EXIT_FAILURE);
+    }
+
+  //9. start writing to the file by creating the file in append mode
+
+  file_fd = open(FILE_STORAGE_PATH,O_APPEND | O_WRONLY); // open file in append mode and write the data received from the client in it
+  if(file_fd == -1)
+    {
+      printf("File open error whilvalgrinde appending\n");
+      syslog(LOG_ERR, "File open error while appending");
+      exit(1);
+    }
+  else
+    {
+      printf("File open succeed while appending\n");
+      syslog(LOG_INFO, "File open succeed while appending");
+    }
 
-      memset(buffer,0,BUFFER_SIZE);
-      file_fd = open(FILE_STORAGE_PATH,O_RDONLY); //open the file in readonly mode
-      if(file_fd == -1)
+  file_write = write(file_fd,output_buffer,strlen(output_buffer));
+  if(file_write == -1)
+    {
+      printf("Error! File could not be written\n");
+      syslog(LOG_ERR,"File could not be written!");
+      exit(1);
+    }
+  else if(file_write != strlen(output_buffer))
+    {
+      printf("Error! File is partially written\n");
+      syslog(LOG_ERR,"File is partially written");
+      exit(1);
+    }
+  else
+    {
+      printf("File is written successfully\n");
+      syslog(LOG_INFO,"File is written successfully");
+    }
+
+  printf("Writing data to file %s \n",output_buffer);
+
+  close(file_fd);
+
+  memset(buffer,0,BUFFER_SIZE);
+  file_fd = open(FILE_STORAGE_PATH,O_RDONLY); //open the file in readonly mode
+  if(file_fd == -1)
+    {
+      printf("Error while file reading\n");
+      syslog(LOG_ERR,"Error while file reading");
+      exit(1);
```

Comment:
@Amey2904dash  Is exiting program the right way to clean up? what happens to mutex here? You may end up in deadlock